### ISSUE TITLE
Function didn't get disabled by re-deploying function.json with disabled: true

### ISSUE THREAD

**Issue by mikhailshilkov** [Azure/azure-functions-host#2583]
EventHub-triggered function was running when I re-deployed it from CLI with `disabled=true` set in `function.json`. Even though the function was shown as disabled, the actual processing did not stop.

#### Investigative information

Please provide the following:

- Timestamp: 23 March  2018 around 23:15 UTC
- Function App version (1.0 or 2.0-beta): 1.0
- Function App name: MikhailMassiveScale
- Function name(s) (as appropriate): MapJob
- Region: West EU

#### Repro steps

Provide the steps required to reproduce the problem:

1. I created a Function triggered by Event Hub. Function was part of a Function App with 4 functions in total.
2. I deployed this function to Azure with CLI.
3. Event Hub had several millions events, so the functions was crunching through those events, about 40 instances were live.
4. I changed `disabled` attribute to `true` and redeployed the Function App with CLI, hoping to stop the processing.
5. Deployment went OK. After refreshing the app in the portal, the function was shown as disabled.

#### Expected behavior

Function is not called anymore.

#### Actual behavior

Event Hub was still reporting large amount of outgoing events. App Insights was still showing ~40 instances with requests being processed and CPU being high.

#### Known workarounds

I had to stop Function App from the portal, and then processing finally stopped.

**Comment by mikhailshilkov** [comment]
cc @paulbatum

**Comment by pragnagopa** [comment]
This is not the expected behavior. Needs investigation. Here is another related question [Azure/Azure-Functions/issues/525](https://github.com/Azure/Azure-Functions/issues/525)

**Comment by paulbatum** [comment]
@mikhailshilkov ahh, i notice that you performed this disable action via hand editing function.json.. was this function.json generated by visual studio? Did it have a `configurationSource:"attributes"` set? If so, you have to disable using attributes.

We have actually improved on this experience in V2. In V1 attributes are the only way (for functions written in VS), while in V2 you can disable functions through the portal (it sets app settings which V2 knows how to honor).

**Comment by mikhailshilkov** [comment]
Ah, yes, my `function.json` is generated but I changed `disabled` manually. I think I tried with `Disabled("settingname")` but something didn't work, I can't remember the exact sequence, so I fell back to manual change in `function.json`.

This 2 (or now 3?) ways to disable a function  might be confusing. People are already puzzled when they don't see their input/output bindings in the generated file, so this might be on the same ballpark.

If you think it's not an issue, you may close.

**Comment by paulbatum** [comment]
Yeah, I get that its confusing. The heart of the problem is that webjobs knows how to interpret attributes, and we have code that knows how to generate attributes from function.json. But we can't take attributes that you gave us and then dynamically modify them based on the contents of function.json.

When we fixed it in V2, we did it through app settings because disabling in the function.json file is really painful in that:
- it doesnt work for slots
- your disabled state gets overwritten if you do another deployment.

We didn't backport the change to V1 because it would have been challenging to do so (configuration pipeline is completely different).

I think I can close this issue, but before I do I want to check whether we have any documentation that actually explains this. If not, at least I can open a doc bug.
